File: ios/BiangBiang Hanzi/Views/SectionView.swift
//
//  SectionView.swift
//  BiangBiang Hanzi
//
//  Reusable titled section with a single trailing action button.
//

import SwiftUI

struct SectionView<Content: View>: View {
    let title: String
    let actionLabel: String
    let actionIcon: String
    let action: () -> Void
    @ViewBuilder let content: Content

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(title)
                    .font(.headline)
                Spacer()
                Button(actionLabel, systemImage: actionIcon, action: action)
            }
            content
        }
    }
}

File: android/app/src/main/java/dev/veeso/harakatlens/ui/components/SectionView.kt
package dev.veeso.harakatlens.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ContentCopy
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp

/**
 * Reusable titled section with a single trailing action button.
 */
@Composable
fun SectionView(
    title: String,
    actionLabel: String,
    actionIcon: ImageVector = Icons.Default.ContentCopy,
    onActionClick: () -> Unit,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.Start,
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
            )
            TextButton(onClick = onActionClick) {
                Icon(
                    actionIcon,
                    contentDescription = null,
                    modifier = Modifier.size(18.dp),
                )
                Spacer(Modifier.size(6.dp))
                Text(actionLabel)
            }
        }

        content()
    }
}
